### PR TITLE
feat(conf): add method to remove a key value pair from the configuration

### PR DESCRIPTION
**Pull request by tareknaser360**
## Description

This pull request adds a new function `rm_conf` to the `CLNConf` struct, which allows to remove a configuration field from the object. The function takes a key (a string) as input, and returns an error if the key is not present in the object. If the key is present, the function removes the corresponding field and returns `Ok(())`.

## How Has This Been Tested?
I've added a unit test for the `rm_conf` function, which ensures that it behaves as expected when removing a field from the object. The test checks that the field is removed correctly and that the object is flushed to disk as expected.

### CODE REVIEW

File: conf/src/lib.rs [hunk 1]
```diff
@@ -104,6 +104,18 @@ impl CLNConf {
...
+            self.fields.remove_entry(key);
+        } else {
+            return Err(ParsingError {
+                core: 2,
+                cause: format!("field with `{key}` not present"),
+            });
+        }
+        Ok(())
+    }
+
```
Comment (vincenzopalazzo) [inline conf/src/lib.rs:118]:
```suggestion
    pub fn rm_conf(&mut self, key: &str, value: Option<&str>) -> Result<(), ParsingError> {
        if self.fields.contains_key(key) {
             // 1. get the value from the map
             // 2. is the value is not known see if the value is equal to the previous 
            self.fields.remove_entry(key);
        } else {
            return Err(ParsingError {
                core: 2,
                cause: format!("field with `{key}` not present"),
            });
        }
        Ok(())
    }

```

OK I found a bug, the filed in the config file are stored as <Key, Vec<Values>> this mean that for a `plugin` field you have a single entry inside the hash map but multiple value.

The quest is we want remove all the entries with a `plugin` field? the answer is:
- Yes when the `value` from the parameter is None;
- No, we want remove a specific value with a parameter with `value` that the user specified

This is useful in your case in coffee, you want remove a plugin and this mean that the plugin is formed by key=plugin, value=path of the plugin

File: conf/src/lib.rs [hunk 2]
```diff
@@ -234,6 +246,27 @@ mod tests {
         cleanup_file(path.as_str());
     }
 
+    #[test]
+    fn flush_conf_three() {
+        let path = get_conf_path();
```
Comment (vincenzopalazzo) [inline conf/src/lib.rs:251]: Can we test the case where we want remove all the plugins, and the case when we want remove just a single plugin?

**Review by vincenzopalazzo — CHANGES_REQUESTED** [review]

File: conf/src/lib.rs (hunk 1 above)
Comment (tareknaser360) [reply conf/src/lib.rs]: Yes. I didn't notice.
Thanks for the review.

File: conf/src/lib.rs (hunk 2 above)
Comment (tareknaser360) [reply conf/src/lib.rs:266]: Ok I added another test for removing just one plugin

**Review by vincenzopalazzo — APPROVED** [review]
ACK https://github.com/laanwj/cln4rust/pull/95/commits/a05e1e7f39fa449da38ce294a26b0dd23ccef1bb

**Pull request merged**